pkg/logger/logger.go

```diff
@@ -120,34 +140,152 @@ func (l *Logger) Error(err error) {
 	}
 }
 
-// isLevelEnabled checks if a given log level should be enabled based on the logger's current level
-func (l *Logger) isLevelEnabled(level LogLevel) bool {
-	if l.LogLevel == LogLevelOff {
-		return false
-	}
-	return logLevelOrder[level] >= logLevelOrder[l.LogLevel]
-}
-
+// Trace logs a message at TRACE level if the current log level permits.
 func (l *Logger) Trace(message string) {
 	if l.isLevelEnabled(LogLevelTrace) {
 		l.log(theme.Colors.Info, message)
 	}
 }
 
+// Debug logs a message at DEBUG level if the current log level permits.
 func (l *Logger) Debug(message string) {
 	if l.isLevelEnabled(LogLevelDebug) {
 		l.log(theme.Colors.Info, message)
 	}
 }
 
+// Info logs a message at INFO level if the current log level permits.
 func (l *Logger) Info(message string) {
 	if l.isLevelEnabled(LogLevelInfo) {
 		l.log(theme.Colors.Info, message)
 	}
 }
 
+// Warning logs a message at WARNING level if the current log level permits.
 func (l *Logger) Warning(message string) {
 	if l.isLevelEnabled(LogLevelWarning) {
 		l.log(theme.Colors.Warning, message)
 	}
 }
+
+// Map Atmos log levels to Charmbracelet log levels for direct conversion.
+var atmosToCharmLevelMap = map[string]log.Level{
+	string(LogLevelTrace):   log.DebugLevel, // TODO: Add trace level using charm custom level
+	string(LogLevelDebug):   log.DebugLevel,
+	string(LogLevelInfo):    log.InfoLevel,
+	string(LogLevelWarning): log.WarnLevel,
+	string(LogLevelError):   log.ErrorLevel,
+	string(LogLevelOff):     log.FatalLevel + 1, // Higher than any level
+}
+
+// AtmosLogger wraps the Charmbracelet logger to provide enhanced structured semantic logging.
+type AtmosLogger struct {
+	*log.Logger
+}
+
+// NewAtmosLogger creates a new AtmosLogger with the given options.
+func NewAtmosLogger(out io.Writer, options *log.Options) *AtmosLogger {
+	return &AtmosLogger{
+		Logger: log.NewWithOptions(out, *options),
+	}
+}
+
+// NewDefaultAtmosLogger creates a new AtmosLogger with default settings.
+func NewDefaultAtmosLogger() *AtmosLogger {
+	return &AtmosLogger{
+		Logger: log.Default(),
+	}
+}
+
+// Log implements the core logging functionality with intelligent handling of different input types.
+func (l *AtmosLogger) Log(level log.Level, msgOrErr interface{}, keyvals ...interface{}) {
+	var msg string
+
+	switch v := msgOrErr.(type) {
+	case *errors.AtmosError:
+		msg = v.Message
+
+		if v.Base != nil {
+			keyvals = append([]interface{}{"base_error", v.Base.Error()}, keyvals...)
+		}
+
+		if len(v.Meta) > 0 {
+			for k, val := range v.Meta {
+				keyvals = append(keyvals, k, val)
+			}
+		}
+
+		if len(v.Tips) > 0 {
+			keyvals = append(keyvals, "tips", v.Tips)
+		}
+
+	case error:
+		msg = v.Error()
+		keyvals = append([]interface{}{"error", v.Error()}, keyvals...)
+
+	case string:
+		msg = v
+
+	default:
+		msg = fmt.Sprintf("%v", v)
+	}
+
+	// This is basically for the case where we have an odd number of key-value pairs
+	if len(keyvals)%2 != 0 {
```

Comment:
IMO this is a red flag. What does
Charm log do in this case? This is silencing a coding mistake, calling the function the wrong way.